header-verifier/src/lib.rs

```diff
@@ -67,6 +67,11 @@ pub trait Headerverifier: bls_signature::BlsSignatureModule {
             "Only ESDT Safe contract can call this endpoint"
         );
 
+        require!(
```

Comment:
Please remove this and keep it like it was before. When the time comes and we will look for possible security issues and audits we'll cover edge cases for this function as well, but for now it's good without this require.

Reply:
As my development cycle, every endpoint that I will write I will also try to cover all the edge cases with `requires`. Afterwards through testing I will add further `requires`. I thought that having a more verbose error message will help in the grand scheme of the execution phase. In this case I will remove it and more `requires` when we realize that they are actually needed.